Flavorly/Views/AddEditRecipeView.swift

```diff
@@ -16,162 +16,192 @@ struct AddEditRecipeView: View {
     @State private var newRecipe = false
     
     @State private var title = ""
-    @State private var type = NSLocalizedString("type.main", comment: "")
     @State private var ingredients = ""
     @State private var text = ""
     @State private var author = ""
-    @State private var diet = NSLocalizedString("diet.omnivore", comment: "")
-    @State private var occasion = NSLocalizedString("occasion.everyday", comment: "")
-    
-    let diets = ["diet.vegetarian",
-                 "diet.vegan",
-                 "diet.noGluten",
-                 "diet.noDairy",
-                 "diet.pescatarian",
-                 "diet.omnivore",
-                 "diet.noNut",
-                ]
-    let types = ["type.soup",
-                 "type.salad",
-                 "type.main",
-                 "type.dessert",
-                 "type.side",
-                 "type.breakfast",
-                 "type.lunch",
-                 "type.dinner",
-                 "type.juice",
-                 ]
-    let occasions = ["occasion.christmas",
-                     "occasion.newYear",
-                     "occasion.birthday",
-                     "occasion.easter",
-                     "occasion.everyday",
-                    ]
+    @State private var tags: [Tag] = []
+    @State private var tagTitle = ""
     
     var hasValidName: Bool {
-        if title.isEmpty || author.isEmpty {
+        if title.isEmpty || author.isEmpty || ingredients.isEmpty || text.isEmpty {
```

Comment:
I think maybe we don't need to worry about the text being empty?
```suggestion
        if title.isEmpty || author.isEmpty || ingredients.isEmpty {
```